mypy/checker.py

```diff
@@ -1012,7 +1012,13 @@ def check_override(self, override: FunctionLike, original: FunctionLike,
         # Use boolean variable to clarify code.
         fail = False
         if not is_subtype(override, original, ignore_pos_arg_names=True):
-            fail = True
+            if (isinstance(override, Overloaded) and
+                    isinstance(original, Overloaded) and
+                    name not in nodes.reverse_op_methods.keys()):
+                # Allow subtype overloads to be greater than their supertype.
+                fail = is_subtype(original, override, ignore_pos_arg_names=True)
```

Comment:
@sixolet 

> So, I read this block as "if they're both overloaded functions, and the override isn't a subtype of the original, it's still ok as long as they're completely incomparable. If the original is a subtype of the override, then it's an error". That doesn't make sense to me.

*cough* I think I might have written that code backwards... *cough*